### PR TITLE
refactor!: rename get options and make get calls network or local only

### PR DESCRIPTION
**Pull request by jost-s**
### Summary
Rename `GetOptions` variants. `Content` becomes `Local` and queries now consistently only locally available data. No network calls are made. This applies to `get` and `get_details`. `Latest` becomes `Network` and fetches from the network if the caller is not an authority of the requested Action or Entry.

There are tests for both calls, using action and entry hash.

I've renamed `Latest` to `Network` as a separate commit, so it's easy to revert in case there are objections. I find it makes more sense with these two options of `Local` vs `Network`.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_zome_types/src/entry.rs [hunk 1]
```diff
@@ -49,58 +49,52 @@ pub struct AppEntryDefLocation {
...
 
 #[derive(PartialEq, Debug, Clone, Serialize, Deserialize)]
-/// Options for controlling how get works
+/// Options for controlling how get is executed.
 pub struct GetOptions {
-    /// If this is true the get call will wait for
-    /// the latest data before returning.
-    /// If it is false you will get whatever is locally
-    /// available on this conductor.
+    /// Set if data should be fetched from the network or only from the local
```
Comment (ThetaSinner) [inline crates/holochain_zome_types/src/entry.rs:54]:
```suggestion
    /// Configure whether data should be fetched from the network or only from the local
```

**Review by ThetaSinner — APPROVED** [review]
Nice, looks great! I have a few minor comments but nothing that would stop me approving :)

File: crates/holochain_cascade/src/lib.rs [hunk 2]
```diff
@@ -693,26 +693,17 @@ where
...
-        let authoring = self.am_i_authoring(&entry_hash.clone().into())?;
-        let authority = self.am_i_an_authority(entry_hash.clone().into()).await?;
         let query: GetEntryDetailsQuery = self.construct_query_with_data_access(entry_hash.clone());
-
-        // We don't need metadata and only need the content
-        // so if we have it locally then we can avoid the network.
-        if let GetStrategy::Content = options.strategy {
+        if let GetStrategy::Local = options.strategy {
+            // Only return what is in the database.
             let results = self.cascading(query.clone()).await?;
```
[the remark this answers is not in the document]
Comment (jost-s) [reply crates/holochain_cascade/src/lib.rs:699]: `cascading` just queries all databases and there's never any network call made. Yes, `fetch_record` is the cascade method to fetch from network if network access is available.

[the remark this answers is not in the document]
Comment (ThetaSinner) [reply crates/holochain_cascade/src/lib.rs]: Got it, thanks for the refresher, I don't know this code very well :) So yes, this looks good then!

**Review by maackle — APPROVED** [review]
Nice clean change, this seems right

**Pull request merged**